server/src/main/java/com/cloud/template/TemplateAdapterBase.java

```diff
@@ -284,35 +285,62 @@ public TemplateProfile prepare(RegisterTemplateCmd cmd) throws ResourceAllocatio
 
     }
 
-    @Override
-    public TemplateProfile prepare(GetUploadParamsForTemplateCmd cmd) throws ResourceAllocationException {
+    /**
+     * Prepare upload parameters internal method for templates and ISOs local upload
+     */
+    private TemplateProfile prepareUploadParamsInternal(boolean isIso, long userId, String name, String displayText,
```

Comment:
can't we use a common parameter object to both commands instead of this insanely long parameterlist(, please)?

Reply:
I could have used an object of the `TemplateProfile` class as parameter, but I think that can be confusing, as it is intended that the method returns a valid `TemplateProfile`. The one passed as parameter will just be a holder for the needed parameters and could cause confusion with the valid one. Do you agree?

Reply:
It is confusing any which way :( also this private method has javadoc which, if needed is not ideal.
I was thinking of putting both cmd in a hierarchy and passing the parent here. or derive the iso cmd from the template one.
anyway not a showstopper, but prone to improvement.

Reply:
Ok, how about now?

Reply:
outstanding 🗡